### PR TITLE
Document docker targets

### PR DESCRIPTION
**Pull request by Martchus**
See particular commit messages

### CODE REVIEW

File: docs/Contributing.asciidoc [hunk 1]
```diff
@@ -374,12 +392,19 @@ Replace VAR1 and VAR2 in -e switch to match a test battery of the test matrix:
...
 |============================
 
-==== tips
+So by replacing VAR1 and VAR2 with those values one can trigger the different tests of the matrix.
+
+Of course it is also possible to run (specific) tests directly via `prove` instead of using the Makefile targets.
 
-Running commands will be executed after the initialization script (database creation and so on..). So if there's the need to run an interactive
-session after it just do:
+You might have noticed that there is also the Makefile target `docker-test`. It also invokes
```
Comment (okurz) [inline docs/Contributing.asciidoc]: I would refrain from noting the actual dependencies of make targets within the documentation. Rather rename the targets in the Makefile to be obvious.

File: Makefile [hunk 2]
```diff
@@ -87,6 +87,7 @@ ifneq ($(CHECKSTYLE),0)
 	PERL5LIB=lib/perlcritic:$$PERL5LIB perlcritic lib
 endif
 
+# runs the tests (using Docker on Travis)
```
[the remark this answers is not in the document]
Comment (Martchus) [reply Makefile]: ok, I'll remove it

File: docs/Contributing.asciidoc [hunk 3]
```diff
@@ -335,37 +335,55 @@ variable +OPENQA_BASEDIR+. The default value is +/var/lib+.
 
 == Running tests of openQA itself
 
-There's two ways of executing the testsuite locally:
```
[the remark this answers is not in the document]
Comment (Martchus) [reply docs/Contributing.asciidoc]: I think no description is better than the description which was there before. But I can write a better one.

**Comment by Martchus** [comment]
@okurz Fixed all points you mentioned.

File: Makefile [hunk 4]
```diff
@@ -123,20 +127,25 @@ public/favicon.ico: assets/images/logo.svg
...
 	env | grep -E 'FULLSTACK|UITEST|GH|TRAVIS|CPAN|DEBUG|ZYPPER' > $(docker_env_file)
 
+# launches a Docker container and runs the testsuite within, does NOT ensure that container is built
 .PHONY: docker-test-run
 docker-test-run: docker.env
 	docker run --env-file $(docker_env_file) -v $(current_dir):/opt/openqa -v /var/run/dbus:/var/run/dbus \
 	   $(DOCKER_IMG) make travis-codecov
 	rm $(docker_env_file)
 
+# launches a Docker container and runs the testsuite within (used on Travis)
```
Comment (okurz) [inline Makefile]: that's just the same as above again!!!1 I would ditch the commend and name it explicitly, e.g. "container-test-build-and-run"

File: Makefile [hunk 5]
```diff
@@ -123,20 +127,25 @@ public/favicon.ico: assets/images/logo.svg
...
 .PHONY: docker-test-build
 docker-test-build:
 	docker build --no-cache $(current_dir)/docker/openqa -t $(DOCKER_IMG)
 
+# prepares the environment variables for running tests within the Docker container
 .PHONY: docker.env
 docker.env:
 	env | grep -E 'FULLSTACK|UITEST|GH|TRAVIS|CPAN|DEBUG|ZYPPER' > $(docker_env_file)
 
+# launches a Docker container and runs the testsuite within, does NOT ensure that container is built
```
[the remark this answers is not in the document]
Comment (okurz) [reply Makefile]: Well, I can't think of anything better so

+1

But maybe strip the "-it" :)

**Comment by Martchus** [comment]
@okurz Added another commit to rename the targets and also removed the comments again.

**Pull request merged**